k256/benches/ecdsa.rs

```diff
@@ -44,14 +44,22 @@ fn bench_ecdsa(c: &mut Criterion) {
     let z = test_scalar_z();
 
     group.bench_function("try_sign_prehashed", |b| {
-        b.iter(|| d.try_sign_prehashed(k, &z).unwrap())
+        b.iter(|| {
+            black_box(d)
```

Comment:
Perhaps I've been using it wrong, but I've always placed `black_box` around the entire operation where I wanted to avoid elision by the optimizer

Reply:
I think ideally you need `black_box` for both inputs and output. The `black_box` on the output is [always there](https://docs.rs/criterion/latest/src/criterion/bencher.rs.html#88), so we don't need to add it explicitly.

The one on the inputs prevents the compiler to optimize on the knowledge about inputs e.g. that we pass the same point/scalar as two different inputs. See here a tiny [example](https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=adacb7a1a4ffd30b541f76d35af697d5) that I built for myself just now. The documentation also [mentions](https://bheisler.github.io/criterion.rs/book/getting_started.html) how to use `black_box` with inputs. I don't think we really needed this property everywhere, but finding out where we need it would take time. So since a few more `black_box`es don't hurt either compilation time or readability significantly, I just added them everywhere.